File: js-demo/payment-demo.js
/**
 * TrigGuard JavaScript Demo
 * Minimal example demonstrating TG.execute() and guard()
 */

// Note: In a real project: import { execute, guard } from "@trigguard/sdk";

// Original payment function (before TrigGuard)
async function sendPayment(amount, recipient) {
  console.log(`💰 Sending ${amount} to ${recipient}`);
}

// Example 1: Using execute() directly
async function executeExample() {
  const receipt = await execute({
    action: "send_payment",
    payload: { amount: 100, recipient: "merchant@example.com" },
    context: {
      userId: "user_123",
      sessionId: "session_456",
      environment: "prod",
      requestId: "req_789"
    }
  });
  console.log(receipt.outcome === "executed" ? "✅ Payment executed" : "🔇 Payment not executed");
}

// Example 2: Using guard() wrapper (invisible integration)
async function guardExample() {
  const receipt = await guard(
    {
      action: "send_payment",
      payload: { amount: 100, recipient: "merchant@example.com" },
      context: { userId: "user_123", sessionId: "session_456", environment: "prod", requestId: "req_789" }
    },
    async () => { await sendPayment(100, "merchant@example.com"); }
  );
  console.log(`Outcome: ${receipt.outcome}`);
}

File: swift-demo/PaymentDemo.swift
import Foundation
import TrigGuardSDK

// Example: Wrapping a payment function with TrigGuard

// Original payment function (before TrigGuard)
func sendPayment(amount: Double, recipient: String) async {
    print("💰 Sending \(amount) to \(recipient)")
}

// Guarded payment function (after TrigGuard)
func guardedSendPayment(amount: Double, recipient: String, context: TGContext) async throws {
    let paymentData = try JSONEncoder().encode(["amount": amount, "recipient": recipient])
    
    // Wrap with TrigGuard - closure only executes if outcome is "executed"
    let receipt = try await TG.guard(
        action: "send_payment",
        payload: paymentData,
        context: context
    ) {
        await sendPayment(amount: amount, recipient: recipient)
    }
    
    // Receipt always returned (silence is not an error)
    switch receipt.outcome {
    case .executed: print("✅ Payment executed")
    case .silent: print("🔇 Payment not executed (expected, not an error)")
    }
}

// Example usage
func main() async throws {
    let context = TGContext(
        userId: "user_123",
        sessionId: "session_456",
        environment: .production,
        requestId: "req_789"
    )
    try await guardedSendPayment(amount: 100.0, recipient: "merchant@example.com", context: context)
}
